File: app/src/main/java/com/pluu/webtoon/item/DetailView.java
package com.pluu.webtoon.item;

/**
 * 상세데이터 타입
 * Created by nohhs on 2015-03-13.
 */
public class DetailView {
	private final VIEW_TYPE type;
	private final String value;
	private final ChatView chatValue;

	public DetailView(VIEW_TYPE type, String value) {
		this.type = type;
		this.value = value;
		this.chatValue = null;
	}

	public DetailView(VIEW_TYPE type, ChatView value) {
		this.type = type;
		this.value = null;
		this.chatValue = value;
	}

	public VIEW_TYPE getType() {
		return type;
	}

	public String getValue() {
		return value;
	}

	public ChatView getChatValue() {
		return chatValue;
	}

	public static DetailView createImage(String value) {
		return generate(VIEW_TYPE.IMAGE, value);
	}

	public static DetailView createText(String value) {
		return generate(VIEW_TYPE.TEXT, value);
	}

	public static DetailView createChatNotice(String value) {
		return new DetailView(VIEW_TYPE.CHAT_NOTICE, new ChatView(null, null, value));
	}

	public static DetailView createChatNoticeImage(String img) {
		return new DetailView(VIEW_TYPE.CHAT_NOTICE_IMAGE, new ChatView(img, null, null));
	}

	public static DetailView createChatLeft(String img, String name, String text) {
		ChatView item = new ChatView(img, name, text);
		return new DetailView(VIEW_TYPE.CHAT_LEFT, item);
	}

	public static DetailView createChatRight(String img, String name, String text) {
		ChatView item = new ChatView(img, name, text);
		return new DetailView(VIEW_TYPE.CHAT_RIGHT, item);
	}

	public static DetailView createChatEmpty() {
		return generate(VIEW_TYPE.CHAT_EMPTY, null);
	}

	private static DetailView generate(VIEW_TYPE type, String value) {
		return new DetailView(type, value);
	}

	@Override
	public String toString() {
		return "DetailView{" +
			"type=" + type +
			", value='" + value + '\'' +
			'}';
	}
}

File: app/src/main/java/com/pluu/webtoon/item/DetailView.kt
package com.pluu.webtoon.item

/**
 * 상세데이터 타입
 * Created by pluu on 2017-05-02.
 */
data class DetailView(val type: VIEW_TYPE, val value: String? = null, val chatValue: ChatView? = null) {
    var height: Float = 0.toFloat()

    companion object {

        fun createImage(value: String): DetailView {
            return generate(VIEW_TYPE.IMAGE, value)
        }

        fun createChatNotice(value: String): DetailView {
            return DetailView(VIEW_TYPE.CHAT_NOTICE, chatValue = ChatView(null, null, value))
        }

        fun createChatNoticeImage(img: String): DetailView {
            return DetailView(VIEW_TYPE.CHAT_NOTICE_IMAGE, chatValue = ChatView(img, null, null))
        }

        fun createChatLeft(img: String, name: String, text: String): DetailView {
            return DetailView(VIEW_TYPE.CHAT_LEFT, chatValue = ChatView(img, name, text))
        }

        fun createChatRight(img: String, name: String, text: String): DetailView {
            return DetailView(VIEW_TYPE.CHAT_RIGHT, chatValue = ChatView(img, name, text))
        }

        fun createChatEmpty(): DetailView {
            return DetailView(VIEW_TYPE.CHAT_EMPTY)
        }

        fun generate(type: VIEW_TYPE, value: String?): DetailView {
            return DetailView(type, value)
        }
    }
}
